### PR TITLE
auto migrate db on connect

### PR DESCRIPTION
**Pull request by nopcoder**
- configuration database.disable_auto_migrate to turn off
- migration script wrap with BEGIN/COMMIT when needed
- fix ctl usage output

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -112,6 +112,7 @@ func (c *Config) GetDatabaseParams() dbparams.Database {
 		MaxOpenConnections:    viper.GetInt("database.max_open_connections"),
 		MaxIdleConnections:    viper.GetInt("database.max_idle_connections"),
 		ConnectionMaxLifetime: viper.GetDuration("database.connection_max_lifetime"),
+		DisableAutoMigrate:    viper.GetBool("database.disable_auto_migrate"),
```
Comment (ozkatz) [inline config/config.go:115]: @nopcoder fine with this - but it needs to be documented.
I recommend setting an explicit default value as well.

This seems to be the case for all other dbparams except for connection string.

**Review by ozkatz — CHANGES_REQUESTED** [review]
LGTM except for missing docs and default values.

File: config/config.go (hunk 1 above)
Comment (nopcoder) [reply config/config.go:115]: Updated the docs. Note that the defaults are set by the 'db' package by `normalizeDBParams` function

Comment (ozkatz) [reply config/config.go:115]: great, thanks!

**Review by ozkatz — APPROVED** [review]
LGTM!

File: config/config.go (hunk 1 above)
Comment (nopcoder) [reply config/config.go:115]: Thanks you!

**Pull request merged**